djangocms_navigation/views.py

```diff
@@ -0,0 +1,91 @@
+from django.apps import apps
+from django.contrib.contenttypes.models import ContentType
+from django.contrib.sites.shortcuts import get_current_site
+from django.core.exceptions import PermissionDenied
+from django.http import JsonResponse
+from django.views.generic import ListView
+
+from cms.models import Page
+
+from djangocms_navigation.utils import is_model_supported, supported_models
+
+
+class ContentObjectSelect2View(ListView):
+    def get(self, request, *args, **kwargs):
+
+        self.site = get_current_site(request)
+        self.object_list = self.get_queryset()
+        context = self.get_context_data()
+        data = {
+            "results": [
+                {"text": str(obj), "id": obj.pk} for obj in context["object_list"]
```

Comment:
Why aren't you just doing `for obj in self.get_queryset()` btw? Since you're not returning a template response (but a json response), you don't actually need context data for anything AFAIK?

Reply:
Agreed. I changed method name to get_data to make it clear and also used View as base class.